### ISSUE TITLE
Drag&drop with floating windows doesn't work

### ISSUE THREAD

**Issue by DimaKoltun** [ocornut/imgui#5710]
----

**Version/Branch of Dear ImGui:**

Version: 1.88 WIP
Branch: docking

**Back-end/Renderer/Compiler/OS**

Back-ends: imgui_impl_sdl.cpp + imgui_impl_opengl3.cpp
Compiler: MSVC
Operating System: Windows 10

**My Issue/Question:**

If we drag item from from floating window that isn't fully located under main window, then drop to another window doesn't work

**Screenshots/Video**

https://user-images.githubusercontent.com/83829508/192271041-2de05418-8661-4d10-a634-a4a4f7566e45.mp4

**Standalone, minimal, complete and verifiable example:** _(see https://github.com/ocornut/imgui/issues/2261)_
```
        ImGui::Begin("Asset Browser");
        if (ImGui::Button("Drag this"))
        {

        }
        int dndData = 1;
        if (ImGui::BeginDragDropSource(ImGuiDragDropFlags_None))
        {
            ImGui::SetDragDropPayload("DND_TEST", &dndData, sizeof(int));
            ImGui::EndDragDropSource();
        }
        ImGui::End();

        ImGui::Begin("Viewport");
        ImGui::Text("Dropped count: %d", droppedCount);
        if (ImGui::Button("Drop Here"))
        {

        }
        if (ImGui::BeginDragDropTarget())
        {
            if (const ImGuiPayload* payload = ImGui::AcceptDragDropPayload("DND_TEST"))
            {
                droppedCount += *(const int*)payload->Data;
            }

            ImGui::EndDragDropTarget();
        }
        ImGui::End();
```

**Comment by ocornut** [comment]
It appears to be a bug in the SDL backend.
We added relaxed suppot for `io.BackendFlags |= ImGuiBackendFlags_HasMouseHoveredViewport;` in 1338eb31, however the Hovered Viewport reported by SDL in this situation is wrong.

A temporary workaround you can use is to call `io.BackendFlags &= ~ImGuiBackendFlags_HasMouseHoveredViewport;` after `ImGui_ImplSDL2_InitXXX()`. Will investigate further.

**Comment by ocornut** [comment]
I experimented and couldn't get SDL to send any `SDL_WINDOWEVENT` event for the underlying window, even when disabling the `SDL_CaptureMouse()` flag.

I narrowed it down to a new feature of SDL 2.0.22:
https://wiki.libsdl.org/SDL_CaptureMouse
> Please note that as of SDL 2.0.22, SDL will attempt to "auto capture" the mouse while the user is pressing a button; this is to try and make mouse behavior more consistent between platforms, and deal with the common case of a user dragging the mouse outside of the window. This means that if you are calling [SDL_CaptureMouse](https://wiki.libsdl.org/SDL_CaptureMouse)() only to deal with this situation, you no longer have to (although it is safe to do so). If this causes problems for your app, you can disable auto capture by setting the [SDL_HINT_MOUSE_AUTO_CAPTURE](https://wiki.libsdl.org/SDL_HINT_MOUSE_AUTO_CAPTURE) hint to zero.

I am pushing a fix to the backend now, need to call `SDL_SetHint(SDL_HINT_MOUSE_AUTO_CAPTURE, "0");`

**Comment by ocornut** [comment]
The full fix is 51bbc70 and has been merged in docking.
Thanks for the careful bug report and repro.

**Comment by ocornut** [comment]
Previous fixes caused issue, pushed a new fix d7aca168, see #5012 for details.
